spec/index.html

```diff
@@ -648,27 +648,26 @@ <h3>IRIs</h3>
         Interoperability problems can be avoided by minting
         only IRIs that are normalized according to
         <a data-cite="rfc3987#section-5">Section 5</a>
-        of [[!RFC3987]]. Non-normalized forms that are best avoided
-        include the following:</p>
+        of [[!RFC3987]]. 
+        </p>
 
       <ul>
-        <li>Uppercase characters in scheme names and domain names</li>
-        <li>Percent-encoding of characters where it is not
-          required by IRI syntax</li>
-        <li>Explicit inclusion of the HTTP default port
-          (<code>http://example.com:80/</code>);
-          <code>http://example.com/</code> is preferable</li>
-        <li>A completely empty path in an HTTP IRI
-          (<code>http://example.com</code>);
-          <code>http://example.com/</code> is preferable</li>
-        <li>“<code>/./</code>” or “<code>/../</code>” in the path
-          component of an IRI</li>
-        <li>Lowercase hexadecimal letters within percent-encoding
-          triplets (i.e., “<code>%3F</code>” is preferred over
-          “<code>%3f</code>”)</li>
-        <li>Punycode-encoding of Internationalized Domain Names
-          in IRIs [[RFC3492]]</li>
-        <li>IRIs that are not in Unicode [=Normalization Form C=] [[I18N-Glossary]]</li>
+        <li>Use lowercase characters in scheme names and domain names.</li>
```

Comment:
Some care needs to be used here. IDNs are not necessarily case-insensitive outside the ASCII range. Mapping the U-label to lowercase might result in a domain name that does not work.

```suggestion
        <li>Use lowercase ASCII characters in scheme names and, where applicable, domain names.
           Non-ASCII domain names should be chosen carefully.</li>
```

Reply:
Just to be clear, we’re using UTF-8, but limited to the ASCII range – we’re not using an ASCII character set.

Reply:
We could split this into scheme advice and domain name advice if they have different text.

I agree with @gkellogg's point about UTF-8/ASCII . For an advice note, is it better to say "ASCII" or is there Unicode-terminology - I found is "Basic Latin" but I don't think that is helpful.

== _scheme name advice_

Scheme names are restricted by syntax anyway so "lowercase" is unambiguous.

With or without "ASCII" --

_text:_  * Use lowercase ASCII characters in [scheme names](https://datatracker.ietf.org/doc/html/rfc3986#section-3.1).

== _domain name advice_

For an advice note, "chosen carefully", is a bit non-committal :smile:. 

What is the state of play in DNS names? Does the upper/lower case advice apply to characters outside basic latin?

_text:_  * Use lowercase ASCII characters in domain names in preference to uppercase ASCII characters.

Possible link: The https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.2 but the key text is not exactly at that point.

Reply:
So between making this comment last night and this morning I double checked the state of IDNA wrt non-ASCII case-sensitivity to ensure I am giving the proper advice. 

Basically, ASCII letters in domain names are case-insensitive (and guidance to lowercase these is the right course of action). Non-ASCII characters are case-sensitive, because upper and lowercase letters encode differently into Punycode and because case folding is at least sometimes locale-sensitive. This is all in RFC5890.

However, RFC5892, which says which code points can be registered, includes section 2.2, which says (in the most oblique and indirect way possible) that uppercase letters (among others) should be avoided.

You're right that scheme names are already restricted to ASCII, so stating ASCII for them is unnecessary (except, perhaps, as a reminder that the letters are restricted to ASCII??). 

For domain names, _only the ASCII letters_ are case-insensitive. If a domain name is registered, in spite of the guidance noted above, with an uppercase non-ASCII letter, that name will encode to a different A-label (and is thus distinct in DNS).

> For an advice note, "chosen carefully", is a bit non-committal 😄.

Yeah, that's an exceptionally lame bit of guidance.

> For an advice note, is it better to say "ASCII" or is there Unicode-terminology - "Basic Latin", but I don't think that is helpful.

ASCII is usually the clearest way to talk about this. "Basic Latin" is Unicode-ese, but only Unicadetti will make the connection to "code points less than 0x80"

Perhaps having two bullets would simplify matters?

```suggestion
        <li>Use lowercase characters in scheme names.</li>
        <li>Use lowercase characters in domain names. 
         Note that non-ASCII characters in domain names are case-sensitive [[RFC5890]], 
         but that domains are generally only registered with lowercase letters [[RFC5892]].</li>
```

Reply:
Part of #155.